APRandomizer/Classes/Archipelago_GameMod.uc

```diff
@@ -1450,6 +1450,8 @@ function LoadSlotData(JsonObject json)
 	SlotData.NoPaintingSkips = json.GetBoolValue("NoPaintingSkips");
 	SlotData.CTRLogic = json.GetIntValue("CTRLogic");
 	SlotData.DeathLink = json.GetBoolValue("death_link");
+	SlotData.DeathLinkAmnesty = json.GetIntValue("death_link_amnesty");
+	SlotData.DWDeathLinkAmnesty = json.GetIntValue("dw_death_link_amnesty");
```

Comment:
I'm not familiar enough with Unreal, does GetIntValue default to something if the key isn't found, or does it raise an exception? If it raises an exception, I'd recommend formatting this in such a way that it doesn't so that it maintains backwards compatibility with multiworlds generated using an APWorld without these options.

though also I'd recommend just having the option be entirely client-side, but that's a personal preference sort of deal

Reply:
I mentioned in the description that they seem to default to 0 which means every death, i.e. the same as old behavior of the mod. Looking at the function source, it calls GetStringValue, which returns an empty string if not found, and then that will get cast to int which gives 0 for non-int values. So this behavior should be consistent instead of relying on an uninitialized read or something.
I added it as a player option mainly since that seems to be what other games with this option do. I also think it makes sense to be able to change the option for amnesty in the same place as death link itself, although this mod does also have a command to toggle it in game.